### DISCUSSION TITLE
compiling SOIL2 with mingw64 gcc version 13.1.0 in windows 11

### DISCUSSION THREAD

**Discussion by svatox** [SpartanJ/SOIL2#64] [Q&A]
I am trying to compile SOIL2 with mingw64 in windows 11
I have installed premake5 and used the following commands:

```bash
.\premake5.exe gmake2
cd make\windows
mingw32-make
```

and it occured errors as following:

```bash
"==== Building soil2-static-lib (debug_x86_64) ===="
Creating ../../obj/windows/x86_64/debug/soil2-static-lib
SOIL2.c
process_begin: CreateProcess(NULL, cc -MD -MP -DDEBUG -m64 -g -Wall -o ../../obj/windows/x86_64/debug/soil2-static-lib/SOIL2.o -MF ../../obj/windows/x86_64/debug/soil2-static-lib/SOIL2.d -c ../../src/SOIL2/SOIL2.c, ...) failed.
make (e=2): 系统找不到指定的文件。
mingw32-make[1]: *** [soil2-static-lib.make:186: ../../obj/windows/x86_64/debug/soil2-static-lib/SOIL2.o] Error 2
mingw32-make: *** [Makefile:45: soil2-static-lib] Error 2
```

I checked the file `../../src/SOIL2/SOIL2.c`, and this file existed. I have no idea about solving the problem. I really need your help, please.

**Comment by SpartanJ** [comment]
It seems that the file that it's not found is the compiler `cc`. There are many distributions of `mingw64` so the problem is specific to some of this distributions, can you tell me how did you install mingw64? Most probable fix is to set the CC and CXX environment variables with your compiler file names.

**Comment by SpartanJ** [comment]
As far as I can remember the distribution that I was using is this one: https://github.com/skeeto/w64devkit
That might help.

**Comment by sherlocknieh** [comment]
**Mystery Solved!**

### 1.You were definitely using Powershell or CMD to execute those commands.
But the makefiles generated by premake5 are for bash, so there will be a lot of path format problems.

**Solution**: Try use git-bash instead.

### 2. You were probably using mingw64 standard version, so there is no a "cc.exe" file.

**Solution**: Just make a copy of "gcc.exe" and rename it to "cc.exe".
(better also copy "mingw32-make to "make".)

### 3. All the steps above done and worked, now you might face a SDL library missing problem.

![image](https://github.com/SpartanJ/SOIL2/assets/43476643/1ca93254-668e-4723-8eff-abf590dac156)

**Solution**: Download [SDL2-devel-2.30.2-mingw.zip](https://github.com/libsdl-org/SDL/releases/download/release-2.30.2/SDL2-devel-2.30.2-mingw.zip), and copy the "x86_64-w64-mingw32" folder and merge to "<yourpath>/mingw64/x86_64-w64-mingw32"
